doc/manual/variables-and-scoping.rst

```diff
@@ -117,24 +128,29 @@ changed within their global scope and not from an outside module.
 Note that the interactive prompt (aka REPL) is in the global scope of
 the module ``Main``.
 
+Within the global scopes, the `global` keyword is never necessary,
+although allowed.
+
+
 .. _man-local-scope:
 
 Local Scope
 -----------
 
-A new local scope is introduced by most code-blocks, see above
-:ref:`table <man-scope-table>` for a complete list.  A local scope
-*usually* inherits all the variables from its parent scope, both for
-reading and writing.  There are two subtypes of local scopes, hard and
-soft, with slightly different rules concerning what variables are
-inherited.  Unlike global scopes, local scopes are not namespaces,
+A new local scope is introduced by most code blocks (see above
+:ref:`table <man-scope-table>` for a complete list).
+A local scope inherits all the variables from a parent local scope,
+both for reading and writing.
+Additionally, the local scope inherits all globals that are assigned
+to in its parent global scope block (if it is surrounded by a global `if` or `begin` scope).
```

Comment:
Does this mean there is a difference between

```
x = 1
function f()
    x = 2
end
```

and

```
begin
    x = 1
    function f()
        x = 2
    end
end
```

? If so, I don't think we want that.